kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/Util.kt

```diff
@@ -72,4 +72,15 @@ private fun <T> renderTree(node: T, visitor: TreeVisitor<T>, indent: String, isL
 
 fun AstType.isSet(): Boolean = packageName == "kotlin.collections" && simpleName == "Set"
 fun AstType.isMap(): Boolean = packageName == "kotlin.collections" && simpleName == "Map"
-fun AstType.isPair(): Boolean = packageName == "kotlin" && simpleName == "Pair"
\ No newline at end of file
+fun AstType.isPair(): Boolean = packageName == "kotlin" && simpleName == "Pair"
+fun AstType.isFunctionOrTypeAliasOfFunction(): Boolean = isFunction() || isTypeAlias() &&
+    resolvedType().isFunctionOrTypeAliasOfFunction()
+
+tailrec fun AstType.resolveToHighestTypeAlias(): AstType {
```

Comment:
This always returns the concrete type correct? I think the name is a bit confusing, maybe something like `fullyResolvedType()`? This could also go on the ast class instead of as an extension but don't have strong opinions about that.